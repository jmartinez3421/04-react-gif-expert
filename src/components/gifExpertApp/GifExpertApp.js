import React, {useState} from 'react';
import AddCategory from '../addCategory/AddCategory';
import GifGrid from '../gifGrid/GifGrid';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Piece']);

    // const handleAdd = () => {
    //     /*Para agregar un nuevo elemento a un array del useState crearemos un nuevo array con [] y dentro de este 
    //     copiaremos el array del useState con ...categories y a√±adiremos el elemento que queramos detras de la ,*/
    //     setCategories( [...categories, 'HunterXHunter']);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr></hr>


            <ol>
                {
                    categories.map(category => {
                        return <GifGrid key={category} category={category} categories={categories} setCategories={setCategories} />;
                    })
                }
                
            </ol>
        </>
    );
}

export default GifExpertApp;
