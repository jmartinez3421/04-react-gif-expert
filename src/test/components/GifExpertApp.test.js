import React from 'react';
import "@testing-library/jest-dom";
import { shallow } from 'enzyme';
import GifExpertApp from "../../components/GifExpertApp";

describe('Pruebas sobre el componente <GifExpertApp />', () => {
    test('El componente debe hacer match con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})