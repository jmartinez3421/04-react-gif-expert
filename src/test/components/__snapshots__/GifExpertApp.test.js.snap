// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifExpertApp /> El componente debe hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas sobre el componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categories={
        Array [
          "One Punch",
          "Dragon ball",
        ]
      }
      category="One Punch"
      key="One Punch"
      setCategories={[Function]}
    />
    <GifGrid
      categories={
        Array [
          "One Punch",
          "Dragon ball",
        ]
      }
      category="Dragon ball"
      key="Dragon ball"
      setCategories={[Function]}
    />
  </ol>
</Fragment>
`;
